Packs/AggregatedScripts/Scripts/QuarantineFile/QuarantineFile.py

```diff
@@ -0,0 +1,1145 @@
+                )
+        return final_results
+
+
+def handler_factory(brand: str, orchestrator) -> BrandHandler:
+    """
+    Factory function that returns an instance of the correct brand handler.
+
+    This allows the orchestrator to dynamically select the appropriate logic
+    based on the brand name discovered for a group of endpoints.
+
+    Args:
+        brand (str): The name of the brand.
+        orchestrator (QuarantineOrchestrator): The orchestrator instance.
+
+    Returns:
+        BrandHandler: An instance of a concrete BrandHandler subclass (e.g., XDRHandler).
+
+    Raises:
+        ValueError: If no handler is available for the specified brand.
+    """
+    demisto.debug(f"[Factory] Creating handler for brand: '{brand}'")
+    if brand in [BRAND_CORE_IR, BRAND_XDR_IR]:
+        demisto.debug("[Factory] Selected XDRHandler.")
+        return XDRHandler(brand, orchestrator)
+    raise ValueError(f"No handler available for brand: {brand}")
+
+
+""" SCRIPT ORCHESTRATOR """
+
+
+class QuarantineOrchestrator:
+    """Manages the entire quarantine lifecycle from start to finish."""
+
+    CONTEXT_PENDING_JOBS = "quarantine_pending_jobs"
+    CONTEXT_COMPLETED_RESULTS = "quarantine_completed_results"
+    ENDPOINT_IDS_ARG = "endpoint_id"
+    FILE_HASH_ARG = "file_hash"
+    FILE_PATH_ARG = "file_path"
+    BRANDS_ARG = "quarantine_brands"
+
+    HASH_TYPE_TO_BRANDS = {"sha256": [BRAND_CORE_IR, BRAND_XDR_IR]}
+
+    def __init__(self, args: dict):
+        """
+        Initializes the QuarantineOrchestrator.
+
+        This involves loading the current state (pending jobs and completed results)
+        from the incident context to support polling.
+
+        Args:
+            args (dict): The arguments passed to the script.
+        """
+        demisto.debug("[Orchestrator] Initializing.")
+        self.args = args
+        self.verbose = argToBoolean(args.get("verbose", False))
+        self.verbose_results: list[CommandResults] = []
+        self.pending_jobs = demisto.get(demisto.context(), self.CONTEXT_PENDING_JOBS) or []
+        # Load results from context, ensuring they are dictionaries
+        self.completed_results: list[QuarantineResult] = [
+            QuarantineResult(**res) for res in (demisto.get(demisto.context(), self.CONTEXT_COMPLETED_RESULTS) or [])
+        ]
+        demisto.debug(
+            f"[Orchestrator] Loaded state. Pending jobs: {len(self.pending_jobs)}, "
+            f"Completed results: {len(self.completed_results)}"
+        )
+
+    def _verify_and_dedup_endpoint_ids(self):
+        """
+        Verifies that endpoint IDs are provided and removes duplicates.
+
+        Returns:
+            list: A list of unique endpoint IDs.
+
+        Raises:
+            DemistoException: If the 'endpoint_id' argument is missing.
+        """
+        if not self.args.get(self.ENDPOINT_IDS_ARG):
+            raise DemistoException(f"Missing required argument. Please provide '{self.ENDPOINT_IDS_ARG}'.")
```

Comment:
i think DemistoException is bad practice - lets use valueerror

Reply:
shouldnt we have an exception then for missing argument?
`MissingArgDemistoException` or something like that?

Reply:
added custom exception

Reply:
resolved?